wix/puppet.wxs

```
@@ -614,6 +614,58 @@
         Name="RememberedPuppetAgentStartupMode" Type="raw" Win64="no" />
     </Property>
 
+    <!-- Properties for existing Ini Values - expanding the remembered properties pattern just a bit
+    -->
+
+    <CustomAction Id="SetIniPropertyValues"
+                  Script="vbscript"
+                  Execute="firstSequence">
+      <![CDATA[
+Function SetPropertyFromIni (keyName, propertyName, textToSearch)
+  Set iniValueRegex = New RegExp
+  iniValueRegex.IgnoreCase = true
+  iniValueRegex.Multiline = true
+  iniValueRegex.Global = false
+  iniValueRegex.Pattern = "^" & keyName & "=(.*[^\s\r\n]*)$"
+
+  Set matches = iniValueRegex.Execute(textToSearch)
+
+  dim iniValue
+
+  If matches.Count <> 0 Then
+    If matches.Item(0).SubMatches.Count <> 0 Then
+      iniValue = matches.Item(0).SubMatches(0)
+      iniValue = Replace(Replace(Replace(iniValue, Chr(13), ""), vbNewLine, ""), vbLf, "")
+    End If
+  End If
+
+  If iniValue <> Empty Then
+    'WScript.Echo "Value for " & keyName & " of '" & iniValue & "' will be set to " & propertyName
+    Session.Property(propertyName) = iniValue
+  End If
+End Function
+
+Dim fso, wshShell, iniPath
```

Not sure if you want to trap or ignore errors from here on (On Error Resume ....)